transport/http/src/main/java/org/kaazing/gateway/transport/http/HttpProtocolCodecFilter.java

```diff
@@ -58,8 +60,23 @@ public void filterWrite(NextFilter nextFilter,
                 IoBufferEx cachedProtocolBuffer = message.getCache().get(nextProtocol);
                 if (cachedProtocolBuffer != null) {
                     if(cachedProtocolBuffer.capacity()==0) {
-                        writeRequest.getFuture().setWritten();
-                        nextFilter.messageSent(session, writeRequest);
+                        // We must commit this http session if the child is committing, to ensure
+                        // the http headers get written
+                        HttpSession child = HttpAcceptor.SESSION_KEY.get(session);
```

Comment:
SESSION_KEY is parametric he typed so I know child is an instance of DefaultHttpSession. But I know nothing about session ( which could be tcp, ssl, etc). I could cast to HttpAcceptSession instead if you think that is better.